src/cpu/utils.hh

```diff
@@ -94,6 +94,22 @@ isAnyActiveElement(const std::vector<bool>::const_iterator& it_start,
     return (it_tmp != it_end);
 }
 
+/**
+ * Get size of inactive tail in an enablement range (0 if none).
+ */
+inline unsigned
+inactiveTailSize(const std::vector<bool>::const_iterator& it_start,
+                 const std::vector<bool>::const_iterator& it_end)
+{
+    std::vector<bool>::const_reverse_iterator rit_start(it_end);
+    std::vector<bool>::const_reverse_iterator rit_end(it_start);
+
+    auto it_tmp = rit_start;
+    unsigned count = 0;
+    for (;it_tmp != rit_end && !(*it_tmp); ++it_tmp, count++);
```

Comment:
haha, whoa. I've got to ask you to re-write this line to be readable as a human. This reminds me of the [obfuscated c competition](https://en.wikipedia.org/wiki/International_Obfuscated_C_Code_Contest)